examples/cutting_stock_problem.rs

```diff
@@ -0,0 +1,179 @@
+/// reason to care for the decision variable `y_j`. To get rid of both, consider the set of patterns
+/// `{p = (p_1, ..., p_m) |  \sum_{i=1..m} w_i p_{i} <= W} = conv({p_t}_{t\in T})`.
+/// We end up with the following (aggregated) master problem
+///
+/// ```
+///   \min \sum_t λ_t * 1
+///      s.t. \sum_t λ_t z_{i,t} >= b_i    [π_i]
+///           λ_t >= 0
+/// ```
+///
+/// where `λ_t` and `z_{i,t}` carry similar meanings as before. A more correct way to write the objective might be to write
+/// `\sum_t λ_t y_t,` but since `y_t` is always `1` that would be redundant. Similarly, the constraint `\sum_t λ_{j,t} = 1` simplifies
+/// to `\sum_t λ_{t} <= k` which is artificial anyway.
+///
+/// # Restricted Master Problem and Pricing Problem
+///
+/// To practically solve the master problem, we need to restrict the set of all patterns to a reasonably small set. The result is a
+/// restricted master problem. A reasonable set of patterns to start with is the set of all patterns that contain exactly one item.
+///
+/// The optimal solution to a restricted master problem is a feasible solution to the original problem. Taking an associated restricted dual
+/// solution, it may or may not be the case that the dual solution is feasible for the original problem. If it is, we would have proven
+/// optimality to the linear problem. If it is not, there is at least one dual constraint that is violated. The dual of the master problem
+/// is
+///
+/// ```
+///   \max \sum_i b_i π_i
+///     s.t. \sum_i π_i z_{i,t} <= 1    [λ_t]
+///          π_i >= 0
+/// ```
+///
+/// Note that the constraint corresponding to `λ_t` is violated iff its slack, known as its reduced cost
+/// `\bar{c}(λ_t) = 1 - \sum_i π_i z_{i,t}`, is negative. Of course, instead of checking for all `t`, we introduce the pricing problem
+///
+/// ```
+///   \min 1 - \sum_i π_i z_{i}
+///     s.t. \sum_i w_i z_{i} <= W
+///          z_{i} >= 0
+/// ```
+///
+// Following https://scipbook.readthedocs.io/en/latest/bpp.html and https://www.gerad.ca/fr/papers/G-2024-36.pdf
+fn main() {
+    let stock_length = 9;
+    let item_sizes = &[6.0, 5.0, 4.0, 2.0, 3.0, 7.0, 5.0, 8.0, 4.0, 5.0];
+    let demand = &[2, 3, 4, 4, 2, 2, 2, 2, 2, 1];
+
+    // Vector of cutting_patterns, initially populated with the trivial ones that contain exactly
+    // one item. cutting_patterns[i][j] indicates how often item j is in pattern i.
+    let initial_cutting_patterns: Vec<Vec<i32>> = (0..item_sizes.len()).map(|i|
+        (0..item_sizes.len()).map(|j| if i == j { 1 } else { 0 }).collect::<Vec<i32>>()
+    ).collect();
+
+    let mut main_problem = Model::default().minimize();
+
+    let cutting_pattern_vars: Vec<Variable> = initial_cutting_patterns.iter().enumerate().map(|(i, &ref _pattern)| {
+        let pattern = (0..10).map(|x| if x == i { "1" } else { "0" }).collect::<Vec<_>>().join("-");
+        main_problem.add_var(0.0, f64::INFINITY, 1.0, &format!("pattern_{pattern}"), VarType::Integer)
```

Comment:
Very optional, but It would be great if you'd use the new ergonomic API here :) 
```suggestion
        main_problem.add(var()int(0..).obj(1.0).name(&format!("pattern_{pattern}"))
```

Reply:
done